src/Pim/Bundle/CatalogBundle/spec/Elasticsearch/Filter/FamilyFilterSpec.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace spec\Pim\Bundle\CatalogBundle\Elasticsearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyTypeException;
+use PhpSpec\ObjectBehavior;
+use Pim\Bundle\CatalogBundle\Elasticsearch\Filter\AbstractFieldFilter;
+use Pim\Bundle\CatalogBundle\Elasticsearch\Filter\FamilyFilter;
+use Pim\Bundle\CatalogBundle\Elasticsearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Exception\InvalidOperatorException;
+use Pim\Component\Catalog\Query\Filter\FieldFilterInterface;
+use Pim\Component\Catalog\Query\Filter\Operators;
+
+/**
+ * Family filter spec for an Elasticsearch query
+ *
+ * @author    Anaël Chardan <anael.chardan@akeneo.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class FamilyFilterSpec extends ObjectBehavior
+{
+    function let() {
+        $this->beConstructedWith(
+            ['family'],
+            ['IN', 'NOT IN' ,'EMPTY', 'NOT EMPTY']
+        );
+    }
+
+    function it_is_initializable() {
+        $this->shouldHaveType(FamilyFilter::class);
+    }
+
+    function it_is_a_filter()
+    {
+        $this->shouldImplement(FieldFilterInterface::class);
+        $this->shouldBeAnInstanceOf(AbstractFieldFilter::class);
+    }
+
+    function it_supports_operators() {
+        $this->getOperators()->shouldReturn([
+            'IN',
+            'NOT IN',
+            'EMPTY',
+            'NOT EMPTY'
+        ]);
+        $this->supportsOperator('EMPTY')->shouldReturn(true);
+        $this->supportsOperator('DOES NOT CONTAIN')->shouldReturn(false);
+    }
+
+    function it_supports_family_field()
+    {
+        $this->supportsField('family')->shouldReturn(true);
+        $this->supportsField('a_not_supported_field')->shouldReturn(false);
+    }
+
+    function it_adds_a_filter_with_operator_in_list(SearchQueryBuilder $sqb) {
+        $sqb->addFilter(
+            [
+                'terms' => [
+                    'family' => ['familyA']
+                ]
+            ]
+        )->shouldBeCalled();
+
+        $this->setQueryBuilder($sqb);
+        $this->addFieldFilter('family', Operators::IN_LIST, ['familyA'], null, null, []);
+    }
+
+    function it_adds_a_filter_with_operator_not_in_list(SearchQueryBuilder $sqb) {
+        $sqb->addMustNot(
+            [
+                'terms' => [
+                    'family' => ['familyA']
+                ]
+            ]
+        )->shouldBeCalled();
+
+        $this->setQueryBuilder($sqb);
+        $this->addFieldFilter('family', Operators::NOT_IN_LIST, ['familyA'], null, null, []);
```

Comment:
What happens if the fourth and fifth parameters are not empties?
The last parameter is an array, perhaps method "options"? Which are the allowed options?